File: Data Structures & Algorithms/valid-binary-search-tree/submission-1.cs
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     public int val;
 *     public TreeNode left;
 *     public TreeNode right;
 *     public TreeNode(int val=0, TreeNode left=null, TreeNode right=null) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

public class Solution {
    public bool IsValidBST(TreeNode root) {

        return ValidNode(root, -1001, 1001);
    }

    private bool ValidNode(TreeNode curr, int lB, int rB)
    {
        if (curr == null)
            return true;

        if (curr.val <= lB || curr.val >= rB)
            return false;
        
        return ValidNode(curr.left, lB, curr.val) && ValidNode(curr.right, curr.val, rB);
    }
}

File: Data Structures & Algorithms/depth-of-binary-tree/submission-1.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def maxDepth(self, root: Optional[TreeNode]) -> int:
        return self.trackDepth(root, 0)

    def trackDepth(self, node: Optional[TreeNode], depth) -> int:
        if node is None:
            return depth
        
        depth += 1
        return max(self.trackDepth(node.left, depth), self.trackDepth(node.right, depth))
        